samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/api/controller/ShoppingCartsController.java

```diff
@@ -0,0 +1,176 @@
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.media.Schema;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartsController {
+  private final CommandHandler<OpenShoppingCart> handleInitializeShoppingCart;
+  private final CommandHandler<AddProductItemToShoppingCart> handleAddProductItemToShoppingCart;
+  private final CommandHandler<RemoveProductItemFromShoppingCart> handleRemoveProductItemFromShoppingCart;
+  private final CommandHandler<ConfirmShoppingCart> handleConfirmShoppingCart;
+  private final CommandHandler<CancelShoppingCart> handleCancelShoppingCart;
+  private final QueryHandler<GetShoppingCartById, Optional<ShoppingCartDetails>> handleGetShoppingCartById;
+  private final QueryHandler<GetShoppingCarts, Page<ShoppingCartShortInfo>> handleGetShoppingCarts;
+
+  public ShoppingCartsController(
+    CommandHandler<OpenShoppingCart> handleInitializeShoppingCart,
+    CommandHandler<AddProductItemToShoppingCart> handleAddProductItemToShoppingCart,
+    CommandHandler<RemoveProductItemFromShoppingCart> handleRemoveProductItemFromShoppingCart,
+    CommandHandler<ConfirmShoppingCart> handleConfirmShoppingCart,
+    CommandHandler<CancelShoppingCart> handleCancelShoppingCart,
+    QueryHandler<GetShoppingCartById, Optional<ShoppingCartDetails>> handleGetShoppingCartById,
+    QueryHandler<GetShoppingCarts, Page<ShoppingCartShortInfo>> handleGetShoppingCarts) {
+
+    this.handleInitializeShoppingCart = handleInitializeShoppingCart;
+    this.handleAddProductItemToShoppingCart = handleAddProductItemToShoppingCart;
+    this.handleRemoveProductItemFromShoppingCart = handleRemoveProductItemFromShoppingCart;
+    this.handleConfirmShoppingCart = handleConfirmShoppingCart;
+    this.handleCancelShoppingCart = handleCancelShoppingCart;
+    this.handleGetShoppingCartById = handleGetShoppingCartById;
+    this.handleGetShoppingCarts = handleGetShoppingCarts;
+  }
+
+  @PostMapping
+  public ResponseEntity openCart(
+    @RequestBody ShoppingCartsRequests.InitializeShoppingCartRequest request
+  ) throws ExecutionException, InterruptedException, URISyntaxException {
+    var cartId = UUID.randomUUID();
+
+    var command = OpenShoppingCart.of(
+      cartId,
+      request.clientId()
+    );
+
+    return ResponseEntity
+      .created(new URI("api/ShoppingCarts/%s".formatted(cartId)))
+      .eTag(handleInitializeShoppingCart.handle(command).value())
+      .build();
+  }
+
+  @PostMapping("{id}/products")
+  public ResponseEntity addProduct(
```

Comment:
In Java is a bad practice to use raw types, if they are parameterized. 
Even IntelliJ warn you about that.

![image](https://user-images.githubusercontent.com/18637116/159584413-5dde3cbb-8e31-4965-a33e-5d9f18e71a1b.png)

If there is no response body you can do `ReponseEntity<Void>`.

Reply:
Fixed in https://github.com/oskardudycz/EventSourcing.JVM/pull/1/commits/ec0198125a019daa9bdfdc574e7cc51ef7761d6e.